integration-libs/cdc/root/service/cdc-auth-http-header.service.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { AuthHttpHeaderService, GlobalMessageType } from '@spartacus/core';
+import { CdcAuthLogOutService } from '../guards/cdc-auth-logout.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CdcAuthHttpHeaderService extends AuthHttpHeaderService {
+  protected cdcAuthLogOutService = inject(CdcAuthLogOutService);
+  /**
+   * override handleExpiredRefreshToken() method
+   */
+  public handleExpiredRefreshToken(): void {
+    this.authRedirectService.saveCurrentNavigationUrl();
+    this.cdcAuthLogOutService.cdcLogout().finally(() => {
```

Comment:
why we need call logout?

Reply:
in AuthHttpHeaderService， here calls `this.authService.coreLogout()`, this method won't logout cdc customer. So we need to replace it with `this.cdcAuthLogOutService.cdcLogout()`